File: python/meteroid/models/entitlement.py
# this file is @generated
import dataclasses
from datetime import datetime

from ..serialization import BaseModel
from .entitlement_id import EntitlementId
from .entitlement_value import EntitlementValue
from .feature_id import FeatureId


@dataclasses.dataclass
class Entitlement(BaseModel):
    """A raw entitlement row attached to one entity (feature, plan version, add-on, or subscription)."""

    created_at: datetime

    feature_id: FeatureId

    id: EntitlementId

    updated_at: datetime

    value: EntitlementValue

File: go/entitlement.go
// this file is @generated
package meteroid

import "time"

// A raw entitlement row attached to one entity (feature, plan version, add-on, or subscription).
type Entitlement struct {
	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	FeatureId FeatureId `json:"feature_id"`

	Id EntitlementId `json:"id"`

	// RFC 3339 timestamp.
	UpdatedAt time.Time `json:"updated_at"`

	Value EntitlementValue `json:"value"`
}
